llama_index/llms/openai.py

```diff
@@ -40,6 +41,10 @@ class OpenAI(LLM, BaseModel):
     additional_kwargs: Dict[str, Any] = Field(default_factory=dict)
     max_retries: int = 10
 
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        validate_openai_api_key()
```

Comment:
Could you add this for embeddings as well?

Reply:
+1

Embeddings seem to be especially messed up for notebooks. I always have to set both an env var and directly on the openai module 🤔 
 
```
import openai
openai.api_key = "sk-..."
```